pkg/datapath/linux/probes/probes.go

```diff
@@ -386,10 +388,137 @@ func HaveBoundedLoops() error {
 }
 
 // HaveFibIfindex checks if kernel has d1c362e1dd68 ("bpf: Always return target
-// ifindex in bpf_fib_lookup") which is 5.10+. This got merged in the same kernel
-// as the new redirect helpers.
-func HaveFibIfindex() error {
-	return features.HaveProgramHelper(ebpf.SchedCLS, asm.FnRedirectPeer)
+// ifindex in bpf_fib_lookup") which is 5.10+ (and backports). It does so by
+// checking that ifindex has been updated as a result of the RIB lookup even
+// though the bpf_fib_lookup failed (no L2 resolution for the NH)
+func HaveFibIfindex() (err error) {
+	defer func() {
+		if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+			log.WithError(err).Fatal("failed to probe HaveFibIfindex")
+		}
+
+		if errors.Is(err, ebpf.ErrNotSupported) {
+			log.Debug("HaveFibIfindex NOT supported")
+		}
+	}()
+
+	ns, err := netns.New()
+	if err != nil {
+		return fmt.Errorf("create netns: %w", err)
+	}
+	defer ns.Close()
+
+	return ns.Do(func() error {
+		lo, err := netlink.LinkByName("lo")
```

Comment:
Is the loopback interface actually used for something?

Why not attach the program to `lo` instead of creating a dummy?

Reply:
Loopback interfaces are somewhat special in several ways. I wasn't sure if there is a difference in behaviour for `bpf_fib_lookup` for loopbacks. I think a more common use-case for cilium is to attach programs to the vif/netkit interface, so having the probe _as close as the real thing_ as possible

Reply:
Code is liability, so less code is better. If the probe works on a loopback interface, it's preferred over creating extra interfaces.

> I wasn't sure if there is a difference in behaviour for `bpf_fib_lookup` for loopbacks.

Routes added to lo automatically go into the local routing table and get the `host` scope, while routes for other interfaces get the `link` scope by default. I don't think it makes a difference for the probe, but I haven't checked.

> I think a more common use-case for cilium is to attach programs to the vif/netkit interface, so having the probe as close as the real thing as possible.

As close to the real thing would mean creating a veth, not a dummy. Though, for the purpose of a route lookup, this doesn't matter.

Reply:
>Routes added to lo automatically go into the local routing table and get the host scope, while routes for other interfaces get the link scope by default. I don't think it makes a difference for the probe, but I haven't checked.

There are many more details (some come from RFCs). See for instance comments like this:

https://github.com/torvalds/linux/blob/master/net/ipv4/route.c#L2544

And others (not extensive):

* https://github.com/torvalds/linux/blob/master/net/ipv4/fib_frontend.c#L389
* https://github.com/torvalds/linux/blob/master/net/ipv4/route.c#L2502
* https://github.com/torvalds/linux/blob/master/net/ipv4/route.c#L2683
* ...

> As close to the real thing would mean creating a veth, not a dummy. Though, for the purpose of a route lookup, this doesn't matter.

I can move it to veth, if you want

Reply:
Adopted the other 2 changes; I will wait to see if this tests needs to be moved out elsewhere to change it to a veth (albeit dummy and veth are really not that different as opposed to lo vs dummy/veth).

Reply:
@ti-mo Maybe this fell off your radar, but for the specific purposes of this PR does it matter to move this to veth or not? If the PR is generally pushing the code into a good direction I don't think we need to make the changes perfect.

Reply:
> Is the loopback interface actually used for something?

When I locally deleted all the other namespace iteration / route handling logic, the only dependency I found was to initialize a specific register in the program with an IP address. I don't think the actual IP address matters, so it seems like this probe could be considerably simplified by removing the loopback,route,namespace logic. Unless there's something specific about the route add logic that validates the functionality for older kernels? At a glance it just looks like a standard route add.